### ISSUE TITLE
Haddock failure with LTS-9.0

### ISSUE THREAD

**Issue by taphu** [GetShopTV/swagger2#112]
I am experiencing a failure when trying to build the haddocks for my stack-based project which depends on `swagger2`.

It turns out I get the same error when I just try to build the haddocks for `swagger2` in isolation. I believe this might be a different issue than https://github.com/GetShopTV/swagger2/issues/66 because there is nothing about a `haddock: internal error: spliceURL UnhelpfulSpan` in this output. I'm not sure if this is a `swagger2` bug, a stack bug, or a haddock bug, but any help would be appreciated. Thanks!

```
[delta:~/tmp]$ stack unpack swagger2-2.1.4.1
Unpacked swagger2-2.1.4.1 to /home/rowens/tmp/swagger2-2.1.4.1/
[delta:~/tmp]$ cd swagger2-2.1.4.1/
[delta:~/tmp/swagger2-2.1.4.1]$ stack init
Looking for .cabal or package.yaml files to use to init the project.
Using cabal packages:
- swagger2.cabal

Selecting the best among 11 snapshots...

* Matches lts-9.0

Selected resolver: lts-9.0
Initialising configuration using resolver: lts-9.0
Total number of user packages considered: 1
Writing configuration to file: stack.yaml
All done.
[delta:~/tmp/swagger2-2.1.4.1]$ stack --version
Version 1.5.1 x86_64 hpack-0.17.1
[delta:~/tmp/swagger2-2.1.4.1]$ stack exec which -- haddock
/home/rowens/.stack/snapshots/x86_64-linux/lts-9.0/8.0.2/bin/haddock
[delta:~/tmp/swagger2-2.1.4.1]$ stack exec haddock -- --version
Haddock version 2.17.4, (c) Simon Marlow 2006
Ported to use the GHC API by David Waern 2006-2008
[delta:~/tmp/swagger2-2.1.4.1]$ stack build --haddock --haddock-deps
swagger2-2.1.4.1: configure (lib)
[1 of 2] Compiling Main             ( /home/rowens/tmp/swagger2-2.1.4.1/Setup.hs, /home/rowens/tmp/swagger2-2.1.4.1/.stack-work/dist/x86_64-linux/Cabal-1.24.2.0/setup/Main.o )
[2 of 2] Compiling StackSetupShim   ( /home/rowens/.stack/setup-exe-src/setup-shim-mPHDZzAJ.hs, /home/rowens/tmp/swagger2-2.1.4.1/.stack-work/dist/x86_64-linux/Cabal-1.24.2.0/setup/StackSetupShim.o )
Linking /home/rowens/tmp/swagger2-2.1.4.1/.stack-work/dist/x86_64-linux/Cabal-1.24.2.0/setup/setup ...
Configuring swagger2-2.1.4.1...
swagger2-2.1.4.1: build (lib)
Preprocessing library swagger2-2.1.4.1...
[ 1 of 14] Compiling Data.Swagger.SchemaOptions ( src/Data/Swagger/SchemaOptions.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger/SchemaOptions.o )
[ 2 of 14] Compiling Data.Swagger.Internal.Utils ( src/Data/Swagger/Internal/Utils.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger/Internal/Utils.o )
[ 3 of 14] Compiling Data.Swagger.Internal.AesonUtils ( src/Data/Swagger/Internal/AesonUtils.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger/Internal/AesonUtils.o )
[ 4 of 14] Compiling Data.Swagger.Internal ( src/Data/Swagger/Internal.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger/Internal.o )
[ 5 of 14] Compiling Data.Swagger.Lens ( src/Data/Swagger/Lens.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger/Lens.o )
[ 6 of 14] Compiling Data.Swagger.Internal.ParamSchema ( src/Data/Swagger/Internal/ParamSchema.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger/Internal/ParamSchema.o )
[ 7 of 14] Compiling Data.Swagger.ParamSchema ( src/Data/Swagger/ParamSchema.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger/ParamSchema.o )
[ 8 of 14] Compiling Data.Swagger.Declare ( src/Data/Swagger/Declare.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger/Declare.o )
[ 9 of 14] Compiling Data.Swagger.Internal.Schema ( src/Data/Swagger/Internal/Schema.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger/Internal/Schema.o )
[10 of 14] Compiling Data.Swagger.Schema ( src/Data/Swagger/Schema.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger/Schema.o )
[11 of 14] Compiling Data.Swagger.Internal.Schema.Validation ( src/Data/Swagger/Internal/Schema/Validation.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger/Internal/Schema/Validation.o )
[12 of 14] Compiling Data.Swagger.Schema.Validation ( src/Data/Swagger/Schema/Validation.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger/Schema/Validation.o )
[13 of 14] Compiling Data.Swagger.Operation ( src/Data/Swagger/Operation.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger/Operation.o )
[14 of 14] Compiling Data.Swagger     ( src/Data/Swagger.hs, .stack-work/dist/x86_64-linux/Cabal-1.24.2.0/build/Data/Swagger.o )
swagger2-2.1.4.1: haddock
Running Haddock for swagger2-2.1.4.1...
Preprocessing library swagger2-2.1.4.1...
Warning: The documentation for the following packages are not installed. No
links will be generated to these packages: fail-4.9.0.0
Haddock coverage:
 100% (  3 /  3) in 'Data.Swagger.SchemaOptions'
   0% (  0 / 21) in 'Data.Swagger.Internal.Utils'
  Missing documentation for:
    Module header
    swaggerFieldRules (src/Data/Swagger/Internal/Utils.hs:30)
    gunfoldEnum (src/Data/Swagger/Internal/Utils.hs:50)
    jsonPrefix (src/Data/Swagger/Internal/Utils.hs:55)
    parseOneOf (src/Data/Swagger/Internal/Utils.hs:68)
    <+> (src/Data/Swagger/Internal/Utils.hs:76)
    genericMempty (src/Data/Swagger/Internal/Utils.hs:80)
    genericMappend (src/Data/Swagger/Internal/Utils.hs:83)
    GMonoid (src/Data/Swagger/Internal/Utils.hs:86)
     (src/Data/Swagger/Internal/Utils.hs:90)
     (src/Data/Swagger/Internal/Utils.hs:94)
     (src/Data/Swagger/Internal/Utils.hs:98)
     (src/Data/Swagger/Internal/Utils.hs:102)
    SwaggerMonoid (src/Data/Swagger/Internal/Utils.hs:106)
     (src/Data/Swagger/Internal/Utils.hs:114)
     (src/Data/Swagger/Internal/Utils.hs:115)
     (src/Data/Swagger/Internal/Utils.hs:116)
     (src/Data/Swagger/Internal/Utils.hs:118)
     (src/Data/Swagger/Internal/Utils.hs:122)
     (src/Data/Swagger/Internal/Utils.hs:126)
     (src/Data/Swagger/Internal/Utils.hs:131)
  29% (  4 / 14) in 'Data.Swagger.Internal.AesonUtils'
  Missing documentation for:
    Module header
    AesonDefaultValue (src/Data/Swagger/Internal/AesonUtils.hs:81)
    sopSwaggerGenericToEncoding (src/Data/Swagger/Internal/AesonUtils.hs:269)
    sopSwaggerGenericParseJSON (src/Data/Swagger/Internal/AesonUtils.hs:190)
    HasSwaggerAesonOptions (src/Data/Swagger/Internal/AesonUtils.hs:70)
    SwaggerAesonOptions (src/Data/Swagger/Internal/AesonUtils.hs:57)
    mkSwaggerAesonOptions (src/Data/Swagger/Internal/AesonUtils.hs:63)
    saoPrefix (src/Data/Swagger/Internal/AesonUtils.hs:68)
    saoAdditionalPairs (src/Data/Swagger/Internal/AesonUtils.hs:68)
    saoSubObject (src/Data/Swagger/Internal/AesonUtils.hs:68)
  11% ( 25 /238) in 'Data.Swagger.Internal'
  Missing documentation for:
    Module header
     (src/Data/Swagger/Internal.hs:179)
     (src/Data/Swagger/Internal.hs:188)
    hostConstr (src/Data/Swagger/Internal.hs:191)
    hostDataType (src/Data/Swagger/Internal.hs:194)
     (src/Data/Swagger/Internal.hs:197)
    MimeList (src/Data/Swagger/Internal.hs:305)
    mimeListConstr (src/Data/Swagger/Internal.hs:308)
    mimeListDataType (src/Data/Swagger/Internal.hs:311)
     (src/Data/Swagger/Internal.hs:314)
    ParamAnySchema (src/Data/Swagger/Internal.hs:342)
    ParamOtherSchema (src/Data/Swagger/Internal.hs:347)
    swaggerItemsPrimitiveConstr (src/Data/Swagger/Internal.hs:380)
    swaggerItemsObjectConstr (src/Data/Swagger/Internal.hs:383)
    swaggerItemsArrayConstr (src/Data/Swagger/Internal.hs:386)
    swaggerItemsDataType (src/Data/Swagger/Internal.hs:389)
     (src/Data/Swagger/Internal.hs:395)
     (src/Data/Swagger/Internal.hs:404)
     (src/Data/Swagger/Internal.hs:412)
    SwaggerKindType (src/Data/Swagger/Internal.hs:439)
    SwaggerType (src/Data/Swagger/Internal.hs:444)
    swaggerTypeConstr (src/Data/Swagger/Internal.hs:458)
    swaggerTypeDataType (src/Data/Swagger/Internal.hs:461)
    swaggerCommonTypes (src/Data/Swagger/Internal.hs:464)
    swaggerParamTypes (src/Data/Swagger/Internal.hs:467)
    swaggerSchemaTypes (src/Data/Swagger/Internal.hs:470)
    swaggerTypeConstrs (src/Data/Swagger/Internal.hs:473)
     (src/Data/Swagger/Internal.hs:477)
     (src/Data/Swagger/Internal.hs:482)
     (src/Data/Swagger/Internal.hs:487)
    ParamLocation (src/Data/Swagger/Internal.hs:492)
    Format (src/Data/Swagger/Internal.hs:512)
    collectionFormatConstr (src/Data/Swagger/Internal.hs:533)
    collectionFormatDataType (src/Data/Swagger/Internal.hs:536)
    collectionCommonFormats (src/Data/Swagger/Internal.hs:540)
     (src/Data/Swagger/Internal.hs:543)
    ParamName (src/Data/Swagger/Internal.hs:550)
    Schema (src/Data/Swagger/Internal.hs:552)
    ParamSchema (src/Data/Swagger/Internal.hs:583)
    Xml (src/Data/Swagger/Internal.hs:610)
    HttpStatusCode (src/Data/Swagger/Internal.hs:653)
     (src/Data/Swagger/Internal.hs:675)
    HeaderName (src/Data/Swagger/Internal.hs:678)
    Header (src/Data/Swagger/Internal.hs:680)
    Example (src/Data/Swagger/Internal.hs:687)
    exampleConstr (src/Data/Swagger/Internal.hs:690)
    exampleDataType (src/Data/Swagger/Internal.hs:693)
     (src/Data/Swagger/Internal.hs:696)
    ApiKeyParams (src/Data/Swagger/Internal.hs:709)
    OAuth2Flow (src/Data/Swagger/Internal.hs:723)
    OAuth2Params (src/Data/Swagger/Internal.hs:730)
    SecuritySchemeType (src/Data/Swagger/Internal.hs:738)
    SecurityScheme (src/Data/Swagger/Internal.hs:744)
     (src/Data/Swagger/Internal.hs:776)
    Referenced (src/Data/Swagger/Internal.hs:794)
     (src/Data/Swagger/Internal.hs:799)
    URL (src/Data/Swagger/Internal.hs:802)
     (src/Data/Swagger/Internal.hs:808)
     (src/Data/Swagger/Internal.hs:812)
     (src/Data/Swagger/Internal.hs:816)
     (src/Data/Swagger/Internal.hs:820)
     (src/Data/Swagger/Internal.hs:824)
     (src/Data/Swagger/Internal.hs:828)
     (src/Data/Swagger/Internal.hs:832)
     (src/Data/Swagger/Internal.hs:836)
     (src/Data/Swagger/Internal.hs:840)
     (src/Data/Swagger/Internal.hs:844)
     (src/Data/Swagger/Internal.hs:848)
     (src/Data/Swagger/Internal.hs:852)
     (src/Data/Swagger/Internal.hs:856)
     (src/Data/Swagger/Internal.hs:860)
     (src/Data/Swagger/Internal.hs:868)
     (src/Data/Swagger/Internal.hs:869)
     (src/Data/Swagger/Internal.hs:870)
     (src/Data/Swagger/Internal.hs:871)
     (src/Data/Swagger/Internal.hs:872)
     (src/Data/Swagger/Internal.hs:873)
     (src/Data/Swagger/Internal.hs:874)
     (src/Data/Swagger/Internal.hs:875)
     (src/Data/Swagger/Internal.hs:876)
     (src/Data/Swagger/Internal.hs:877)
     (src/Data/Swagger/Internal.hs:879)
     (src/Data/Swagger/Internal.hs:882)
     (src/Data/Swagger/Internal.hs:886)
     (src/Data/Swagger/Internal.hs:890)
     (src/Data/Swagger/Internal.hs:894)
     (src/Data/Swagger/Internal.hs:899)
     (src/Data/Swagger/Internal.hs:909)
     (src/Data/Swagger/Internal.hs:912)
     (src/Data/Swagger/Internal.hs:915)
     (src/Data/Swagger/Internal.hs:918)
     (src/Data/Swagger/Internal.hs:921)
     (src/Data/Swagger/Internal.hs:924)
     (src/Data/Swagger/Internal.hs:927)
     (src/Data/Swagger/Internal.hs:930)
     (src/Data/Swagger/Internal.hs:933)
     (src/Data/Swagger/Internal.hs:936)
     (src/Data/Swagger/Internal.hs:943)
     (src/Data/Swagger/Internal.hs:946)
     (src/Data/Swagger/Internal.hs:949)
     (src/Data/Swagger/Internal.hs:952)
     (src/Data/Swagger/Internal.hs:955)
     (src/Data/Swagger/Internal.hs:958)
     (src/Data/Swagger/Internal.hs:961)
     (src/Data/Swagger/Internal.hs:964)
     (src/Data/Swagger/Internal.hs:967)
     (src/Data/Swagger/Internal.hs:974)
     (src/Data/Swagger/Internal.hs:989)
     (src/Data/Swagger/Internal.hs:993)
     (src/Data/Swagger/Internal.hs:1003)
     (src/Data/Swagger/Internal.hs:1007)
     (src/Data/Swagger/Internal.hs:1011)
     (src/Data/Swagger/Internal.hs:1015)
     (src/Data/Swagger/Internal.hs:1019)
     (src/Data/Swagger/Internal.hs:1026)
     (src/Data/Swagger/Internal.hs:1032)
     (src/Data/Swagger/Internal.hs:1035)
     (src/Data/Swagger/Internal.hs:1039)
     (src/Data/Swagger/Internal.hs:1043)
     (src/Data/Swagger/Internal.hs:1047)
     (src/Data/Swagger/Internal.hs:1051)
     (src/Data/Swagger/Internal.hs:1055)
     (src/Data/Swagger/Internal.hs:1059)
     (src/Data/Swagger/Internal.hs:1063)
     (src/Data/Swagger/Internal.hs:1066)
    referencedToJSON (src/Data/Swagger/Internal.hs:1069)
     (src/Data/Swagger/Internal.hs:1073)
     (src/Data/Swagger/Internal.hs:1074)
     (src/Data/Swagger/Internal.hs:1075)
     (src/Data/Swagger/Internal.hs:1077)
     (src/Data/Swagger/Internal.hs:1087)
     (src/Data/Swagger/Internal.hs:1094)
     (src/Data/Swagger/Internal.hs:1103)
     (src/Data/Swagger/Internal.hs:1116)
     (src/Data/Swagger/Internal.hs:1119)
     (src/Data/Swagger/Internal.hs:1129)
     (src/Data/Swagger/Internal.hs:1132)
     (src/Data/Swagger/Internal.hs:1135)
     (

--  While building package swagger2-2.1.4.1 using:
      /home/rowens/tmp/swagger2-2.1.4.1/.stack-work/dist/x86_64-linux/Cabal-1.24.2.0/setup/setup --builddir=.stack-work/dist/x86_64-linux/Cabal-1.24.2.0 haddock --html --html-location=../$pkg-$version/ --haddock-option=--hyperlinked-source
    Process exited with code: ExitFailure (-11)
[delta:~/tmp/swagger2-2.1.4.1]$
```

**Comment by phadej** [comment]
I have seen the same issue (haddock output just ends, without an error). I have no solution, I moved to use nightly with GHC-8.2, there haddock works. So probably some bug in haddock which is fixed in version bundled with GHC-8.2.

**Comment by fizruk** [comment]
Closing this as outdated. Please reopen if you still experience the issue.
